lightning-invoice/src/utils.rs

```diff
@@ -221,6 +201,91 @@ where
 	}
 }
 
+/// Filters channels for an invoice, and returns the corresponding route hints to include
+/// in the invoice.
+///
+/// The filtering ensures that the `RouteHints` returned only include the highest inbound capacity
+/// channel per counterparty node. If any channel with a higher inbound capacity than the given
+/// `min_inbound_capacity_msat` exists, channels to other counterparty nodes with a lower inbound
+/// capacity than `min_inbound_capacity_msat` will be filtered out, even if they are the highest
+/// inbound capacity channel for that specific counterparty node.
+/// If any public channel exists, the function returns no RouteHints, and the sender will need to
+/// look at the public channels to find a path instead.
+///
+/// Input:
+/// `channels`: The channels to filter.
+/// `min_inbound_capacity_msat`: Defines the lowest inbound capacity channels must have to not
+/// be filtered out, if any other channel above that amount exists.
+///
+/// Result:
+/// `Vec<RouteHint>`: The filtered `RouteHints`, which will be empty if any public channel exists.
```

Comment:
I'd recommend removing this especially if it's repeating what's in the previous paragraphs in a slightly different way. Prefer to say it just once in a succinct way without loss of meaning.